bsip-0040.md

```diff
@@ -0,0 +1,345 @@
+  1. keep the authorities enabled by specifying them in a list of `custom_active_authorities` in `extensions` of `account_update_operation`
+  2. enable them again afterwards by using `update_custom_active_authority`
+* Operation-specific authorities (if present) must be evaluated in incoming transactions
+* Remove expired custom_active_authorities on maintenance if they are expired for longer than one month
+* Additional committee parameters may be needed to limit the extend of usage of this feature
+
+Notes: The implementation must not differentiate on which operation the custom active authority is applied, all operations are treated in same fashion
+
+# Discussion
+
+To be found in the [issue](https://github.com/bitshares/bitshares-core/issues/1061) and [pull request](https://github.com/bitshares/bsips/pull/86).
+
+# Examples
+
+These examples are for illustration and no specification of actual serialization.
+
+
+#### Example: Nested arguments like `options`
+Assume `asset_update_operation`. All attributes of its `options` must be filled on update call. This assert can not be used to realize a "may only change attribute xzy of `options`". This would require that the logic knows which of the arguments are reflected on-chain and it knows how to query it for every operation that contains `options`. If `options` are to be restricted with this assert, all values that should not change would need be fixated by defining an `any` assert for those attributes, while having e.g. a `lt` assert for the one attribute that is allowed to change.
+
+#### Example: Simple transfer
+Assume account A and B and some unrelated key K. 
+The custom active authority should be put such that a transfer transaction sending funds away from A can be signed with key K, as long as the receiver is B. More concrete, the authority would look like
+```
+custom active authority = {
+    valid_from: 7.7.2018 00:00
+    valid_to: 8.7.2018 00:00
+    operation_id: transfer,
+    authority: {
+       threshold: 1
+       key_auth: [key K, 1]
+       account_auth: []
+    },
+    restrictions: [
+        {
+            function: any,
+            argument: to,
+            data: [account B]
+        } // this restricts the argument identified with "to"
+    ]
+}
+```
+Exemplatory outcomes:
+- Transfer asset X with amount Y from account A to account B, signed with Key K: Accepted
+- Transfer asset X with amount Y from account B to account A, signed with Key K: Denied
+- Transfer asset X with amount Y from account A to account C, signed with Key K: Denied
+- Transfer asset X with amount Y from account A to account B, signed with active authority of B: Denied
+- Transfer asset X with amount Y from account A to account B, signed with active authority of A: Accepted
+- Create a proposal that includes operation 'transfer asset X with amount Y from account A to account B', signed with Key K: Denied
```

Comment:
Please remove that line. Authority for the creation of a proposal does not depend on the operations contained therein. Creating a proposal with key K is possible if K has authority to pay fees.

Reply:
That was a moment of forgetting all my BitShares knowledge ... fixed now